test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+      grade_begin: 1, availability: "available", total_copies: 1,
+      call_number: 12345, language: "English", physical_description: "physical_description",
+      primary_language: "English", created_at: created_at, updated_at: updated_at,
+      available_copies: 1, bnumber: "B456", set_type: "set_type",
+      area_of_study: "area_of_study", subjects: [])
+
+      @expected_resp = {:title => "title",:description => "description",:contents => "contents",
+       :id => 123,:details_url => "details_url",:grade_end => 1,:grade_begin => 1,:availability => "available",
+       :total_copies => 1, :call_number => "12345",:language => "English",
+       :physical_description => "physical_description",:primary_language => "English",
+       :created_at => "2014-10-19T01:00:00+0000",:updated_at => "2014-10-19T01:00:00+0000",:available_copies => 1,
+       :bnumber => "B456",:set_type => "set_type",:area_of_study => "area_of_study",:subjects => []}
+    end
+
+    it 'update teacher-set document in es' do
+      resp = nil
+      @mintest_mock1.expect(:call, @expected_resp)
+      @mintest_mock2.expect(:update_document_by_id, @es_doc, [@expected_resp[:id], @expected_resp])
+      
+      @ts_obj.stub :teacher_set_info, @mintest_mock1 do
+        ElasticSearch.stub :new, @mintest_mock2 do
+          resp = @ts_obj.create_or_update_teacherset_document_in_es
+        end
+      end
+
+      @mintest_mock1.verify
+      @mintest_mock2.verify
+      assert_equal(188, resp)
+    end
+
+    it 'test StandardError while updating elastic-search' do
+      resp = nil
+      @mintest_mock1.expect(:call, @expected_resp)
+      @mintest_mock2.expect(:update_document_by_id, @es_doc, [nil, @expected_resp])
+    
+      resp = assert_raises(StandardError) do
+        @ts_obj.stub :teacher_set_info, @mintest_mock1 do
+          ElasticSearch.stub :new, @mintest_mock2 do
+            resp = @ts_obj.create_or_update_teacherset_document_in_es
+          end
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+      @mintest_mock1.verify
+    end
   end
 
 
+  # Delete teacher-set document in ES
+  describe "delete teacherset record from elastic search" do
+    it 'test delete_teacherset_record_from_es method' do
+      expected_resp = {"_index" => "teacherset", "_type" => "teacherset", "_id" => @teacher_set.id, "_version" => 2, "result" => "deleted", 
+        "_shards" => {"total" => 2, "successful" => 1, "failed" => 0}, "_seq_no" => 294, "_primary_term" => 2}
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [@teacher_set.id])
+      resp = nil
+      ElasticSearch.stub :new, @mintest_mock1 do
+        resp = @teacher_set.delete_teacherset_record_from_es
+      end
+      @mintest_mock1.verify
+      assert_equal(184, resp)
+    end
+
+    it 'test StandardError while deleting elastic-search document' do
```

Comment:
This is a good test — basically it's saying that if we get a `nil` response from ES then an error is thrown, right? 

If so would be good to describe that a bit more, something like "TeacherSet#delete_teacherset_record_from_es raises StandardError if ES doesn't return the expected response"